examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+    arguments: [
+      `0x${Hex.fromHexInput(token1Addr).toStringWithoutPrefix()}`,
+      `0x${Hex.fromHexInput(token2Addr).toStringWithoutPrefix()}`,
+      false,
+      "200000",
+      "300000",
+      "200",
+      "300",
+    ],
+  };
+  const result: any = await aptos.view({ payload });
+  console.log("Optimal LP amount: ", result);
+};
+
+const addLiquidity = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
+    data: {
+      function: `0x${swap.toStringLongWithoutPrefix()}::router::add_liquidity_entry`,
+      arguments: [
+        AccountAddress.fromHexInput(token1Addr),
+        AccountAddress.fromHexInput(token2Addr),
+        new Bool(false),
+        new U64(200000),
+        new U64(300000),
+        new U64(200),
+        new U64(300),
+      ],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: deployer, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log("Add liquidity succeed. - ", response.hash);
+  return response.hash;
+};
+
+const swap = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  deployer: Account,
+  fromToken: HexInput,
+  toToken: HexInput,
+  amountIn: number,
+  amountOutMin: number,
+  recipient: AccountAddress,
+): Promise<string> => {
+  const rawTxn = await aptos.generateTransaction({
+    sender: deployer.accountAddress.toString(),
+    data: {
+      function: `0x${swap.toStringLongWithoutPrefix()}::router::swap_entry`,
+      arguments: [
+        new U64(amountIn),
+        new U64(amountOutMin),
+        AccountAddress.fromHexInput(fromToken),
+        AccountAddress.fromHexInput(toToken),
+        new Bool(false),
+        recipient,
+      ],
+    },
+  });
+  const pendingTxn = await aptos.signAndSubmitTransaction({ signer: deployer, transaction: rawTxn });
+  const response = await aptos.waitForTransaction({ transactionHash: pendingTxn.hash });
+  console.log("Swap succeed. - ", response.hash);
+  return response.hash;
+};
+
+const getAssetType = async (aptos: Aptos, owner: Account): Promise<any> => {
+  const data = await aptos.getFungibleAssetMetadata({
+    options: {
+      where: {
+        creator_address: { _eq: owner.accountAddress.toString() },
+      },
+    },
+  });
+
+  if (data.length !== 2) throw new Error("Expected two Fungible Asset.");
+
+  console.log("Fungible Asset: ", data);
+  return {
+    cat: data[0].asset_type,
+    dog: data[1].asset_type,
+  };
+};
+
+const getFaBalance = async (aptos: Aptos, owner: Account, assetType: string): Promise<number> => {
+  const data = await aptos.getCurrentFungibleAssetBalances({
+    options: {
+      where: {
+        owner_address: { _eq: owner.accountAddress.toString() },
```

Comment:
We should note here, I don't think that addresses are stored in shorter mode, making sure that the account addresses match?

Reply:
Or better we could provide an eq implementation that ignores 0x prefix (similar to equalIgnoreCase)?

Reply:
This is GraphQL, so it's direct to the DB, without having a service that fronts the indexer and does conversions, it's going to have to match exactly

Reply:
The query can be modified to do an OR with the WHERE query right?

Reply:
I believe the `owner_address` that stored in DB are always prefixed with `0x`

Reply:
owner_address is stored as a full hex, with a 0x in front.  So, this likely needs to be .toLongString()

Reply:
The `.toString()` here actually coming from Hex.toString() that dport implemented, so it should always contains the `0x` prefix 😄

Reply:
It doesn't just need the `0x`, it actually needs the full `0's` so you'll need to use `toLongString()`, double check with the indexer folks

Reply:
ohh interesting. I thought `toString()` is already converting to long string which contains all of the leading zeros if there any. 

Ok, we probably need to implement a `toLongString()` then.

Reply:
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/core/accountAddress.ts#L142-L144

Reply:
yeah I just realized we already have one on `AccountAddress`, I was looking at the wrong class `Hex` lol all fixed now #134